libs/ml/src/optimisation/adam_optimiser.cpp

```diff
@@ -157,6 +157,55 @@ void AdamOptimiser<T>::ResetCache()
   beta2_t_ = beta2_;
 }
 
+template <class T>
+OperationsCount AdamOptimiser<T>::ChargeConstruct(std::shared_ptr<Graph<T>> graph)
+{
+  // Graph need to be compiled in order to deduce all weight sizes
+  // TODO 2423: Compile shouldn't be needed
+  for (auto &train : graph->GetTrainables())
+  {
+    if (!train->IsInit())
+    {
+      graph->Compile();
+      break;
+    }
+  }
+
+  OperationsCount op_cnt{1};
+  for (auto &train : graph->GetTrainables())
+  {
+    auto     weight    = train->GetWeights();
+    SizeType data_size = TensorType::PaddedSizeFromShape(weight.shape());
+    op_cnt += data_size * 5;
```

Comment:
it's not super great to have an unexplained value (5) here. At least this should be a named constant which hints at what the 5 is doing and why the value is 5.